apm-sniffer/apm-sdk-plugin/spymemcached-2.x-plugin/src/main/java/org/skywalking/apm/plugin/spymemcached/v2/define/MemcachedInstrumentation.java

```diff
@@ -0,0 +1,82 @@
+package org.skywalking.apm.plugin.spymemcached.v2.define;
+
+import static org.skywalking.apm.agent.core.plugin.match.NameMatch.byName;
+
+import java.util.List;
+
+import org.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.ClassInstanceMethodsEnhancePluginDefine;
+import org.skywalking.apm.agent.core.plugin.match.ClassMatch;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+/**
+ * 
+ * {@link MemcachedInstrumentation} presents that skywalking intercept all constructors and methods of 
+ * {@link net.spy.memcached.MemcachedClient}.
+ * {@link XMemcachedConstructorWithInetSocketAddressListArgInterceptor} intercepts the constructor with
+ * argument {@link java.net.InetSocketAddress}.
```

Comment:
Sample comments only:
{@link MemcachedInstrumentation}  represents a skywalking plugin define, which declares to intercept xxxx 's constructors and yyyy's methods.